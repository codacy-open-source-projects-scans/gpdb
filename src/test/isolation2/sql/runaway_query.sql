CREATE EXTENSION IF NOT EXISTS gp_inject_fault;
CREATE OR REPLACE LANGUAGE plpgsql;

CREATE TABLE runaway_query_test_table(a bigint NOT NULL) DISTRIBUTED BY (a);

-- Use error fault to simulate vmem protect error and force cancel query.
SELECT gp_inject_fault_infinite('gpdbwrappers_get_comparison_operator', 'error', 1);

-- Following query will trip the fault during ORCA optimization
EXPLAIN (COSTS OFF) SELECT a FROM runaway_query_test_table WHERE (a = ANY ((ARRAY[42])));

SELECT gp_inject_fault('all', 'reset', 1);
