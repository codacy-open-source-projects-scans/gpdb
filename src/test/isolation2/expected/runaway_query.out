CREATE EXTENSION IF NOT EXISTS gp_inject_fault;
CREATE EXTENSION
CREATE OR REPLACE LANGUAGE plpgsql;
CREATE LANGUAGE

CREATE TABLE runaway_query_test_table(a bigint NOT NULL) DISTRIBUTED BY (a);
CREATE TABLE

-- Use error fault to simulate vmem protect error and force cancel query.
SELECT gp_inject_fault_infinite('gpdbwrappers_get_comparison_operator', 'error', 1);
 gp_inject_fault_infinite 
--------------------------
 Success:                 
(1 row)

-- Following query will trip the fault during ORCA optimization
EXPLAIN (COSTS OFF) SELECT a FROM runaway_query_test_table WHERE (a = ANY ((ARRAY[42])));
 QUERY PLAN                                    
-----------------------------------------------
 Gather Motion 1:1  (slice1; segments: 1)      
   ->  Seq Scan on runaway_query_test_table    
         Filter: (a = ANY ('{42}'::integer[])) 
 Optimizer: Postgres query optimizer           
(4 rows)

SELECT gp_inject_fault('all', 'reset', 1);
 gp_inject_fault 
-----------------
 Success:        
(1 row)
